src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -37,6 +39,57 @@
  */
 public abstract class QueryBuilder<QB extends QueryBuilder> extends ToXContentToBytes implements Writeable<QB> {
 
+    public static enum Operator {
+        OR, AND;
+
+        public BooleanClause.Occur toMustOrShouldClause() {
+            switch (this) {
+                case OR:
+                    return BooleanClause.Occur.SHOULD;
+                case AND:
+                    return BooleanClause.Occur.MUST;
+                default:
+                    throw Operator.newOperatorException(this.toString());
+            }
+        }
+
+        public static Operator readFrom(StreamInput in) throws IOException {
+            switch (in.readVInt()) {
+                case 0:
+                    return OR;
+                case 1:
+                    return AND;
+                default:
+                    throw new ElasticsearchException("Serialized operator needs to be either '1' or '0'");
+            }
+        }
+
+        public void writeTo(StreamOutput out) throws IOException {
+            switch (this) {
+                case OR:
+                    out.writeVInt(0);
+                case AND:
+                    out.writeVInt(1);
+                default:
+                    throw Operator.newOperatorException(this.toString());
+            }
+        }
+
+        public static Operator fromString(String op) {
+            if ("or".equalsIgnoreCase(op)) {
+                return Operator.OR;
+            } else if ("and".equalsIgnoreCase(op)) {
+                return Operator.AND;
+            } else {
+                throw Operator.newOperatorException(op);
+            }
+        }
+        
+        private static ElasticsearchException newOperatorException(String op) {
+            return new ElasticsearchException("Operator needs to be either 'and' or 'or', but not [" + op + "]");
```

Comment:
You could probably make the error more generic by replacing 'and' and 'or' with Lists.newArrayList(Operator.values()) that should give you a nice output of all the permitted values.